decidim-core/lib/decidim/nicknamizable.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module Decidim
+  # This concern contains the logic related to nicknames.
+  module Nicknamizable
```

Comment:
Since we're including validation logic here like `nickname_length_range`, I would turn this into a full-fledged `ActiveSupport::Concern` (or its lowest-level approach, if you like it). 

Also, a bit of documentation regarding its usage & methods wouldn't harm. This is quite a sensible part of the code that might need a bit of understanding.

Reply:
I can document the public methods :+1:.

Regarding the other suggestion, I'm "meh" on it, but I'll do it!